env0/provider_test.go

```diff
@@ -0,0 +1,39 @@
+package env0
+
+import (
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/env0/terraform-provider-env0/utils"
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"testing"
+)
+
+var (
+	apiClientMock *client.MockApiClientInterface
+	ctrl          *gomock.Controller
+)
+
+var testUnitProviders = map[string]func() (*schema.Provider, error){
+	"env0": func() (*schema.Provider, error) {
+		provider := Provider()
+		provider.ConfigureFunc = func(d *schema.ResourceData) (interface{}, error) {
+			return apiClientMock, nil
+		}
+		return provider, nil
+	},
+}
```

Comment:
Can you explain more about this logic? and what is the goal?

Reply:
terraform's `resource.TestCase` expects the get a `providerFactory`. Here what I do is I create the provider factory but I override its `configureFunc` (that is in charge of setting the `meta` object which the API client sits on) to return the mocked API client.